style-guide.md

```diff
@@ -0,0 +1,165 @@
+# edX JavaScript Conventions
+
+# TODO
+
+I'm looking for widespread agreement on everything contained in this document!
+In particular, I'd love feedback on the [Goals](#goals), [Style](#style), and
+[Conventions](#conventions) sections.
+
+Here are a few things that would be awesome to have:
+- [ ] define and link to a jshint/jscs/eslint config
+- [ ] clarify path towards ES2015
+
+## Motivation
+
+Code conventions are important.  It helps us intuitively approach new code,
+revisit old code, and share code with others.
+
+## Goals
+
+We hope to write JavaScript which is:
+
+1. Legible
+2. Friendly
+3. Predictable
+4. Semantic
+
+## About
+
+This guide is split up into two sections - [Style](#style) and
+[Conventions](#conventions).  The style section addresses aesthetic topics,
+while the conventions section addresses topics that have more semantic
+significance.
+
+## Note: ECMAScript Version
+
+Currently, edX only supports ES5.  In order to implement modules, we use AMD by
+way of [RequireJS](http://requirejs.org/).  We are moving towards ES2015
+however, and this guide will be updated when we get there.
+
+## Style
+### Indentation
+
+All JavaScript should be indented with 4 spaces.  edX uses this standard
+throughout our Python, HTML, SASS/CSS, and JavaScript code.
+
+e.g.
+
+```javascript
+function sayHello (name) {
+    return 'Hello, ' + name;
+}
+```
+
+### Semicolons
+
+Use them!
+
+### Line Length
+
+Break lines at `120` columns.
```

Comment:
I believe - though this may no longer be true - that Github's "unified" diff view caps out at 118 characters, so I've used a line length of 117 at previous jobs for this reason